### PR TITLE
feat(station): monitor external canisters

### PR DESCRIPTION
**Pull request by jedna**

### CODE REVIEW

File: core/station/impl/src/factories/requests/monitor_external_canister.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    for MonitorExternalCanisterRequestCreate
+{
+    async fn create(
+        &self,
+        request_id: UUID,
+        requested_by_user: UUID,
+        input: station_api::CreateRequestInput,
+        operation_input: station_api::MonitorExternalCanisterOperationInput,
+    ) -> Result<Request, RequestError> {
+        let request = Request::new(
```
Comment (keplervital) [inline core/station/impl/src/factories/requests/monitor_external_canister.rs:25]: this should now align with the latest format in the main branch

Comment (jedna) [reply core/station/impl/src/factories/requests/monitor_external_canister.rs:25]: yes, this is covered in https://github.com/dfinity/orbit/pull/419/commits/26b8669251c1162e70da964e401ea49507f589ad

File: core/station/impl/src/services/cycle_manager.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#[derive(Debug, Default)]
+pub struct CycleManager {}
+
+impl CycleManager {
+    fn new() -> Self {
+        FUND_MANAGER.with(|manager| {
+            let mut manager = manager.borrow_mut();
+            // Strategy can be default as we always override per canister
+            // Obtain cycles config is inherited from the system service
+            let options = FundManagerOptions::new().with_interval_secs(60 * 60 * 6);
```
Comment (keplervital) [inline core/station/impl/src/services/cycle_manager.rs:32]: nit: make a note here that this interval of 6 hours is used for all monitored canisters

File: core/station/impl/src/services/system.rs [hunk 3]
```diff
@@ -734,39 +687,22 @@ mod install_canister_handlers {
     /// Starts the fund manager service setting it up to monitor the upgrader canister cycles and top it up if needed.
     pub fn monitor_upgrader_cycles(
```
Comment (keplervital) [inline core/station/impl/src/services/system.rs:688]: nit: this method is now doing more then monitoring the upgrader_cycles, maybe rename to `register_cycles_monitor` or similar

File: core/station/impl/src/services/system.rs [hunk 4]
```diff
@@ -734,39 +687,22 @@ mod install_canister_handlers {
...
-            );
-
-            fund_manager.with_options(fund_manager_options);
+        let fund_strategy = FundStrategy::BelowEstimatedRuntime(
+            EstimatedRuntime::new()
+                .with_min_runtime_secs(14 * 24 * 60 * 60) // 14 days
+                .with_fund_runtime_secs(30 * 24 * 60 * 60) // 30 days
+                .with_max_runtime_cycles_fund(1_000_000_000_000)
+                .with_fallback_min_cycles(125_000_000_000)
+                .with_fallback_fund_cycles(250_000_000_000),
```
Comment (keplervital) [inline core/station/impl/src/services/system.rs:698]: we need to account for the cycles reservation here to at least make sure the monitored canister has enough cycles to do an upgrade:

```suggestion
                .with_fallback_min_cycles(400_000_000_000)
                .with_fallback_fund_cycles(200_000_000_000),
```

this change would make sure that the monitored canister has at least 300B cycles in an application subnet to perform a install operation, as well as there is `25%` surplus to account for more consumption

File: apps/wallet/src/components/settings/obtain-cycle/MintFromNativeToken.vue [hunk 5]
```diff
@@ -47,13 +48,25 @@ const model = computed({
   set: value => emit('update:modelValue', value),
 });
 
+const station = useStationStore();
+
+const [icpAsset] = station.configuration.details.supported_assets.filter(asset =>
+  asset.metadata.some(
+    data => data.key === 'ledger_canister_id' && data.value === 'ryjl3-tyaaa-aaaaa-aaaba-cai',
```
Comment (keplervital) [inline apps/wallet/src/components/settings/obtain-cycle/MintFromNativeToken.vue:55]: nit: maybe move those to a variable in the const file

```suggestion
    data => data.key === 'ledger_canister_id' && data.value === 'ryjl3-tyaaa-aaaaa-aaaba-cai',
```

**Review by keplervital — APPROVED** [review]

**Review by keplervital — APPROVED** [review]

**Pull request merged**